core/cloudflow-akka/src/main/scala/cloudflow/akkastream/AkkaStreamletContextImpl.scala

```diff
@@ -73,7 +74,11 @@ final class AkkaStreamletContextImpl(
   }
 
   private val bootstrapServers = system.settings.config.getString("cloudflow.kafka.bootstrap-servers")
-  private def groupId[T](savepointPath: SavepointPath, streamletRef: String, inlet: CodecInlet[T]) = s"${savepointPath.appId}.${streamletRef}.${inlet.name}"
+  private def groupId[T](savepointPath: SavepointPath, streamletRef: String, inlet: CodecInlet[T]) = {
+    val base = s"${savepointPath.appId}.${streamletRef}.${inlet.name}"
+    if (inlet.hasUniqueGroupId) base + randomUUID.toString
+    else base
+  }
```

Comment:
Are there any length restrictions in group IDs we need to worry about here? A GUID is quite long...

Reply:
No:
https://github.com/apache/kafka/blob/72282ed1988e33000f91672a72c7356e26658241/core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala#L799